apps/condo/domains/common/components/PostMessageProvider/globalHandlers.ts

```diff
@@ -138,4 +140,20 @@ export const useUpdateProgressBarHandler: () => RequestHandler<'CondoWebAppUpdat
         return { updated: true }
         // TODO(DOMA-5171): Adding miniAppTaskUIInterface in deps causing rerender hell!
     }, [userId])
+}
+
+export const useRedirectHandler: () => RequestHandler<'CondoWebAppRedirect'> = () => {
+    const router = useRouter()
+
+    return useCallback(({ url, target }) => {
+        if (target === '_blank') {
+            if (typeof window !== 'undefined' && isSafeUrl(url)) {
+                window.open(url, target)
+            }
+        } else {
+            router.push(url)
```

Comment:
I guess checking for safe url should be here too. And also it's seems weird to allow redirect in same window on any provided url....